index.js

```diff
@@ -1,26 +1,19 @@
-const express = require('express');
-const app = express();
-const bodyParser = require('body-parser');
-
-// parse application/x-www-form-urlencoded
-app.use(bodyParser.urlencoded({ extended: false }));
-
-// parse application/json
-app.use(bodyParser.json());
-
-app.use(require('./app/controllers/searchController'));
-
-app.use(require('./app/controllers/airportDetailsController'));
-
-app.use(require('./app/controllers/flightDetailController'));
-
-app.use(require('./app/controllers/seatDetailsController'));
-
-let server = app.listen(8080);
-console.log('Server running in port: 8080 ...');
-
-module.exports = server;
-module.exports.stop = (done) => {
-    server.close();
-    done();
+const searchController = require('./app/controllers/searchController');
+exports.handler = async (event) => {
+    // if there is nothing in querystring, then just pass an empty obj instead of undefined
+    const filter = (event && event.params && event.params.querystring) ? event.params.querystring : {};
+    console.log('below is the filter');
+    console.log(filter);
+
+  const promise = new Promise(function(resolve, reject) {
```

Comment:
This procedure for turning your callback-based API into a Promise looks really clunky to me. (You may have improved it in subsequent commits; I'm working my way through this PR's current state commit-by-commit.)

Reply:
Yes, I agree, but I am not sure how to improve it though.

Reply:
As is so often the case in JS and other languages that are strong in functional programming, higher-order functions are the tool I would reach for first. You can try something like what I did in utopia-airlines/UtopiaAirlinesOnlineAPI@63acce10fb169c419adadf641be89da71b2ff39b: make a method taking the backend method to call as a parameter.